atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableCreator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.cassandra.thrift.Compression;
+import org.apache.thrift.TException;
+
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.protos.generated.TableMetadataPersistence;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.common.base.Throwables;
+import com.palantir.logsafe.SafeArg;
+
+class CassandraTableCreator {
+    private static final String ROW = "key";
+    private static final String COLUMN = "column1";
+    private static final String TIMESTAMP = "column2";
+    private static final String VALUE = "value";
+
+    private final CassandraClientPool clientPool;
+    private final CassandraKeyValueServiceConfig config;
+
+    CassandraTableCreator(CassandraClientPool clientPool, CassandraKeyValueServiceConfig config) {
+        this.clientPool = clientPool;
+        this.config = config;
+    }
+
+    void createTables(Map<TableReference, byte[]> tableRefToMetadata) {
+        try {
+            clientPool.runWithRetry(client -> {
+                for (Map.Entry<TableReference, byte[]> entry : tableRefToMetadata.entrySet()) {
+                    createTable(entry.getKey(), entry.getValue(), client);
+                }
+                CassandraKeyValueServices.waitForSchemaVersions(config, client, "after adding the column family for "
```

Comment:
mentioned in the delete section as well; I think we should be wrapping all calls:
```
for (table in tables) {
    // wait for schema versions
    // create table (or do any schema thing)
    // wait for schema versions
}
```